R/derive_interp_records.R

```diff
@@ -0,0 +1,114 @@
+#' Derive interpolated rows for the CDC charts (>=2 yrs old)
+#'
+#' Derive interpolated rows for the CDC charts (>=2 yrs old) by age in days
+#' for the following parameters: HEIGHT, WEIGHT and BMI
+#'
+#' @param dataset Input metadataset
+#'
+#'   The variables `AGE`, `AGEU`, `SEX`, `L`, `M`, `S` are expected to be in the dataset
+#'
+#'   For BMI the additional variables `P95` and `Sigma` are expected to be in the dataset
+#'
+#'   Note that `AGE` must be in days so that `AGEU` is equal to `"DAYS"`
+#'
+#' @param by_vars Grouping variables
+#'
+#'   The variable from `dataset` which identifies the group of observations
+#'   to interpolate separately.
+#'
+#' @param parameter CDC/WHO metadata parameter
+#'
+#' *Permitted Values*: `"WEIGHT"`, `"HEIGHT"` or `"BMI"` only - Must not be `NULL`
+#'
+#'   e.g. `parameter = "WEIGHT"`,
+#'        `parameter = "HEIGHT"`,
+#'   or   `parameter = "BMI"`.
+#'
+#' @return The input dataset plus additional interpolated records.
+#'
+#' @family metadata
+#'
+#' @keywords metadata
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#'
+#' cdc_htage <- admiralpeds::cdc_htage %>%
+#'   mutate(
+#'     SEX = case_when(
+#'       SEX == 1 ~ "M",
+#'       SEX == 2 ~ "F",
+#'       TRUE ~ NA_character_
+#'     ),
+#'     # Ensure first that Age unit is "DAYS"
+#'     AGE = round(AGE * 30.4375)
+#'   ) %>%
+#'   # Interpolate the AGE by SEX
+#'   derive_interp_records(by_vars = exprs(SEX), parameter = "HEIGHT")
+#'
+#' print(cdc_htage)
+derive_interp_records <- function(dataset,
+                                  by_vars = NULL,
+                                  parameter) {
+  # Apply assertion to by_vars argument to ensure it is appropriate class
+  assert_vars(by_vars, optional = TRUE)
+
+  stopifnot(!is.null(parameter))
+  stopifnot(!is.na(parameter))
+  stopifnot(parameter %in% c("HEIGHT", "WEIGHT", "BMI"))
+
+  metadata_vars <- c("AGE", "L", "M", "S")
+  if (parameter == "BMI") {
+    # metadata_vars <- c("AGE", "L", "M", "S", "P95", "Sigma")
+    metadata_vars <- append(metadata_vars, c("P95", "Sigma"))
+  }
+
+  stopifnot("AGE" %in% colnames(dataset))
```

Comment:
```suggestion
  assert_data_frame(dataset, required_vars = exprs(!!by_vars, AGE, AGEU, L, M, S))
  if (parameter == "BMI") {
    assert_data_frame(dataset, required_vars = exprs(!!by_vars, AGE, AGEU, L, M, S, P95, Sigma))
  }
```
lines 62 - 80 can be handled with something along the lines of this

Reply:
Implemented